docs/sphinx/_ext/sphinx-mdx-builder/sphinxcontrib/mdxbuilder/writers/mdx.py

```
@@ -168,6 +174,55 @@ def __init__(self, document: nodes.document, builder: "MdxBuilder") -> None:
     def add_text(self, text: str) -> None:
         self.states[-1].append((-1, text))
 
+    def get_source_github_url(self, objname: str, modname: str, fullname: str) -> Optional[str]:
+        """Generate a GitHub URL for a Python object.
+
+        Args:
+            objname: Name of the object
+            modname: Module name
+            fullname: Full qualified name
+
+        Returns:
+            A URL to the GitHub source or None if not available
+        """
+        if not self.show_source_links or not modname:
+            return None
+
+        try:
+            module = __import__(modname, fromlist=[""])
+            obj = module
+            parts = fullname.split(".")
+
+            # Navigate to the actual object
+            for part in parts:
+                if hasattr(obj, part):
+                    obj = getattr(obj, part)
+                else:
+                    return None
+
+            if not obj:
+                return None
```

should we raise an exception here ?

---

Added warning